docs/user/library/data/geopackage.rst

```diff
@@ -131,3 +163,25 @@ Tile can then be read back using a ``TileReader``, as follows (the zoom and row/
             }
         }
 
+Using GeoPackage Extensions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The GeoPackage specification is modular using the concepts of extensions to support additional functionality.
+
+* ``GeoPkgExtension`` - base class for geopackage extensions
+* ``GeoPkgExtensionFactory`` - used to advertise additional extensions provided by client code
+
+The GeoPackage module supports the following extensions:
+
+* ``GeoPkgMetadataExtension`` - Uses ``geopkg_metadata`` and and ``geopkg_metadata_reference`` to store metadata.
+* ``GeoPkgSchemaExtension`` - Allows additional description of table columns.
+
+GeoPackageProcessRequest
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+This java bean (and xml bindings) is used to support the GeoServer WPS GeoPackage process which supports the creation of GeoPackages with additional extensions.
+
+* ``GeoPackageProcessRequest.FeatureLayer``
+* ``GeoPackageProcessRequest.TileLayer``
+
+These classes cannot directly be used by GeoTools code.
```

Comment:
Indeed I've always wondered why they are here, rather than in the GeoServer community module.... they were already there when I first met the module...

Reply:
Would like to see these migrate to the geoserver codebase, or the code that processes this data structure moved over to geotools (kind of hard since these data structures reference geoserver layers).

Reply:
I kind of get it now, the "direct" geopackage api takes some of these data structures as a parameters. It is just odd to have a a nice extensible geopackage extension mechanism and then make it closed by demanding everything be controlled by a common data structure.